lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -542,6 +542,43 @@ class MaxPool2dOpConversionPattern
   }
 };
 
+class BroadcastOpConversionPattern
+    : public OpConversionPattern<ttir::BroadcastOp> {
+  using OpConversionPattern<ttir::BroadcastOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(ttir::BroadcastOp srcOp, ttir::BroadcastOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    mlir::Value input = srcOp.getOperand(0);
+    mlir::Value result = srcOp.getResult();
+
+    SmallVector<Operation *> srcOpUsers;
+    for (Operation *nextOp : srcOp->getUsers()) {
+      srcOpUsers.push_back(nextOp);
+    }
+
+    // Try to fold this operation into all Element wise consumer ops.
+
+    for (Operation *nextOp : srcOpUsers) {
+      StringRef opName = nextOp->getName().getStringRef();
+      if (not opName.starts_with("ttir.broadcast") and
+          not nextOp->hasTrait<::mlir::tt::ttir::ElementwiseOp::Trait>()) {
+        return rewriter.notifyMatchFailure(
+            srcOp, "Implicit broadcast operation is only supported for "
+                   "Elementwise consumer ops.");
+      }
+
+      nextOp->replaceUsesOfWith(result, input);
+    }
```

Comment:
I think all of the code above can be replaced with [replaceAllOpUsesWith](https://mlir.llvm.org/doxygen/classmlir_1_1RewriterBase.html#af11a67366e4e1531f3bb0fbe6a67ec03):
```c++
rewriter.replaceAllOpUsesWith(ValueRange(srcOp.getInput()));
rewriter.eraseOp(srcOp);
```

Reply:
Yes I considered it, but I wanted to iterate over the list of Users() so I can verify they are Elementwise ops and produce an error otherwise. 

```
not nextOp->hasTrait<::mlir::tt::ttir::ElementwiseOp::Trait>()) {
        return rewriter.notifyMatchFailure(
            srcOp, "Implicit broadcast operation is only supported for "
                   "Elementwise consumer ops.");
```

Reply:
I think we should just do it wholesale like above, and let each op's verify function assert their own capabilities.  That decouples the pass from being in the business of encoding which op kinds have which abilities.

Reply:
Alright I updated with changes.